src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,129 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+
+    private Node<K, V>[] buckets;
+
+    private int size;
+    private int capacity = DEFAULT_CAPACITY;
+    private double threshold = capacity * LOAD_FACTOR;
+
+    public MyHashMap() {
+        buckets = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
```

Comment:
Since you separated nullable key case from the rest, this method seems a bit swollen.
How about a simplification:
1. Create a method `getIndex(K key)` that will return the index of a given key.
2. Get existing node from data table.
3. Iterate over nodes (`while (node != null); ... node = node.next;`) to check in value should be inserted.
4. If not found, add new node on index.